Analysis/src/ConstraintGraphBuilder.cpp

```diff
@@ -1151,6 +1152,23 @@ ControlFlow ConstraintGraphBuilder::visit(const ScopePtr& scope, AstStatDeclareC
 
     scope->exportedTypeBindings[className] = TypeFun{{}, classTy};
 
+    if (FFlag::LuauParseDeclareClassIndexer && declaredClass->indexer)
+    {
+        RecursionCounter counter{&recursionCount};
+
+        if (recursionCount >= FInt::LuauCheckRecursionLimit)
+        {
+            reportCodeTooComplex(declaredClass->indexer->location);
+        }
+        else
+        {
+            ctv->indexer = TableIndexer{
+                resolveType(scope, declaredClass->indexer->indexType, /* inTypeArguments */ false),
+                resolveType(scope, declaredClass->indexer->resultType, /* inTypeArguments */ false),
+            };
+        }
```

Comment:
Its not completely similar to `checkPack` so i put it as an if-else instead of an early return. I assume we should continue processing the CTV even if the indexer is too complex